File: checkio_js/home/python/secondIndex.py


def secondIndex(text, symbol):
    try:
        return text.index(symbol, text.index(symbol) + 1)
    except:
        return 'undefined'

if __name__ == '__main__':
    print('Example')
    print(secondIndex("sims", "s"))

# These "asserts" are used for self-checking and not for an auto-testing
assert secondIndex("sims", "s") == 3
assert secondIndex("find the river", "e") == 12
assert secondIndex("hi", " ") == 'undefined'
assert secondIndex("hi mayor", " ") == 'undefined'
assert secondIndex("hi mr Mayor", " ") == 5
print("You are awesome! All tests are done! Go Check it!")

File: checkio/elementary/javascript/second_index.js
function secondIndex(text, symbol) {
    var count = 0
    for (let i = 0; i < text.length; i++){
        if (text[i] === symbol){
            count++
            if (count == 2){
                return i
            }
        }
    }
    return undefined
}

console.log(secondIndex("sims", "s"))           // 3
console.log(secondIndex("find the river", "e")) // 12
console.log(secondIndex("hi", " "))             // undefined
console.log(secondIndex("hi mayor", " "))       // undefined
console.log(secondIndex("hi mr Mayor", " "))    // 5
